src/modules/SearchPage/index.tsx

```diff
@@ -1,67 +1,151 @@
+  PokemonTypeDetails,
+  PokemonMoveDetails,
+  PokemonDetails,
+  PokemonData,
+  PokemonMove,
+  PokemonType,
+  MovesData,
+  TypesData,
+  Pokemon,
+  Lengths,
+} from 'types';
 
 type Props = {
   componentName: string;
   location: string;
 };
 
 type State = {
+  moves: Array<PokemonMoveDetails>;
+  types: Array<PokemonTypeDetails>;
   pokemons: Array<PokemonDetails>;
+  activeTab: AvailableTabs;
   inputValue: string;
   isLoading: boolean;
+  lengths: Lengths;
 };
 
 export class SearchPage extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
+      activeTab: AvailableTabs.pokemons,
       isLoading: true,
       inputValue: '',
       pokemons: [],
+      moves: [],
+      types: [],
+      lengths: {
+        moves: 0,
+        pokemons: 0,
+        types: 0,
+      },
     };
   }
+
   componentDidMount() {
-    const value = window.localStorage.getItem(INPUT_VALUE_KEY);
-    if (value != null) {
-      this.setState({ inputValue: value });
+    const inputValue = window.localStorage.getItem(INPUT_VALUE_KEY);
+    if (inputValue && inputValue.trim().length > 0) {
+      this.setState({ inputValue });
+      this.fetchAndSetSpecificData(inputValue).then(() =>
+        setTimeout(() => this.setState({ isLoading: false }), 500)
+      );
+    } else {
+      this.fetchAndSetAllData().then(() =>
+        setTimeout(() => this.setState({ isLoading: false }), 500)
+      );
     }
-    fetchPokemons(API_URL)
-      .then((pokemons) => fetchDetailedPokemons(pokemons))
-      .then((pokemons) => this.setState({ pokemons }))
-      .finally(() => setTimeout(() => this.setState({ isLoading: false }), 1000));
   }
-  onChange = (event: ChangeEvent<HTMLInputElement>) => {
+
+  onClick = (tab: AvailableTabs) => this.setState({ activeTab: tab });
+
+  onChange = async (event: ChangeEvent<HTMLInputElement>) => {
     const { value } = event.target;
     this.setState({ inputValue: value });
     window.localStorage.setItem(INPUT_VALUE_KEY, value);
   };
+
+  onKeyDown = async ({ key }: KeyboardEvent<HTMLInputElement>) => {
+    if (key === 'Enter') {
+      this.setState({ isLoading: true });
+      const value = this.state.inputValue;
+      if (value.trim().length > 0) await this.fetchAndSetSpecificData(value);
+      else await this.fetchAndSetAllData();
+      this.setState({ isLoading: false });
+    }
+  };
+
+  fetchAndSetSpecificData = async (value: string) => {
+    const type = await fetchPokemonByParameter<PokemonTypeDetails>(`${API.TYPE}/${value}`);
+    const move = await fetchPokemonByParameter<PokemonMoveDetails>(`${API.MOVE}/${value}`);
+    const pokemon = await fetchPokemonByParameter<PokemonDetails>(`${API.POKEMON}/${value}`);
+    this.setState({
+      pokemons: pokemon ? [pokemon] : [],
+      types: type ? [type] : [],
+      moves: move ? [move] : [],
+      lengths: {
+        pokemons: pokemon ? 1 : 0,
+        types: type ? 1 : 0,
+        moves: move ? 1 : 0,
```

Comment:
5? 7? 13? 25? 232?
magic numbers

Reply:
These are not magic at all. There are two fetching methods. Fetch all data method will be invoked when input is empty and 3 api calls there return arrays so I have their lengths. But when we type something to the input and search for specific data, then fetch specific data method does 3 api calls that return object. So for example if some pokemon is found, length must equal 1, if not - 0 (pokeapi search for exact value - seems like there's no elastic search used)